test/distributed/fsdp/test_fsdp_comm_hooks.py

```diff
@@ -123,6 +168,8 @@ def test_default_communication_hook_behavior(
 
             # For each worker, the gradient on the weight should be worker_rank.
             grad = net_default_hook.params[0].grad
+            if sharding_strategy != ShardingStrategy.NO_SHARD:
```

Comment:
I am actually curious what the purpose of this assertion is. This seems to be testing if `._is_sharded` is set correctly by FSDP given a sharding strategy.

Reply:
@awgu Rohan asked to make sure that for sharded strategies we actually have a sharded parameter